code/game/objects/items/weapons/grenades/custom_grenades.dm

```diff
@@ -303,3 +303,83 @@
 	new /obj/item/multitool(src) // used to adjust the chemgrenade's signaller
 	new /obj/item/assembly/signaler(src)
 	return
+
+// --------------------------------------
+//  BSA Signaller
+// --------------------------------------
+
+/obj/item/grenade/bsa
+	name = "bluespace artillery signal beacon"
+	desc = "Call in a one time strike with bluespace artillery. It is filled with various biometric security measures."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "batterer"
+	item_state = "batterer"
+	req_one_access = list(access_cent_commander)
+	var/strike_init_time = 8 //in seconds, we multiply by 10 below so we can use this var in the messages we send. so a value of 8 here would be a value of 80 for the timer.
+	var/arm_time = 40 //in deciseconds, so 40 would equal 4 seconds.
+	var/caller
+	var/centcom_cancast = FALSE //incase admin wants to varedit to allow on Z 2 for whatever reason.
+
+/obj/item/grenade/bsa/attack_self()
+	var/mob/user = usr
+	var/turf/bombturf = get_turf(src)
+	if(active)
+		return
+
+	if(is_admin_level(bombturf.z) && !centcom_cancast)
+		to_chat(user, "<span class = 'warning'>The strike craft isn't authorized to fire near Central Command!</span>")
+		return
+
+	if((!ismindshielded(user)) || ((user.mind.special_role != SPECIAL_ROLE_ERT) && (user.mind.special_role != SPECIAL_ROLE_DEATHSQUAD) && (!allowed(user)))) //if is not mindshielded, or is not an ERT/DS AND does not have high level CC access. naval officers have access_cent_commander, but not special role so we don't want to lock them out.
```

Comment:
This check should probably be a sub-proc too, so that subtypes of this item can easily override that check.
I could see us adding a version that's authorized for DS/Specops but not ERT, or for syndicate but not anyone else, etc.

Reply:
then I need to make a proc on the mind/mob level to check their roles

Reply:
Hardcoded checks like this are *strongly* frowned upon in nearly all circumstances, with few exceptions, especially when it comes to using devices. Yes, they do exist else where, but they're typically *edge case*